File: content/_code-samples/freeze/js/check-no-freeze.js
// Dependencies for Node.js.
// In browsers, use <script> tags as in the example demo.html.
if (typeof module !== "undefined") {
  // Use var here because const/let are block-scoped to the if statement.
  var xrpl = require('xrpl')
}


// Connect -------------------------------------------------------------------
async function main() {
  console.log("Connecting to Mainnet...")
  const client = new xrpl.Client('wss://s1.ripple.com')
  await client.connect()

  const my_address = 'rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn';

  // Request account info for my_address to check account settings ------------
  const response = await client.request(
    {command: 'account_info', account: my_address })
  const settings = response.result
  const lsfNoFreeze = xrpl.LedgerEntry.AccountRootFlags.lsfNoFreeze

  console.log('Got settings for address', my_address);
  console.log('No Freeze enabled?',
    (settings.account_data.Flags & lsfNoFreeze) 
    === lsfNoFreeze)

  await client.disconnect()

  // End main()
}

main().catch(console.error)

File: content/_code-samples/freeze/py/check_no_freeze.py
from xrpl.clients import JsonRpcClient
from xrpl.models import AccountInfo

client = JsonRpcClient("https://xrplcluster.com") # Connect to a network


ACCOUNT_ROOT_LEDGER_FLAGS: dict[str, int] = {
        "lsfNoFreeze": 0x00200000,
        "lsfGlobalFreeze": 0x00400000,
    }

def parse_account_root_flags(flags: int) -> list[str]:
    flags_enabled = []
    for flag in ACCOUNT_ROOT_LEDGER_FLAGS:
        check_flag = ACCOUNT_ROOT_LEDGER_FLAGS[flag]
        if check_flag & flags == check_flag:
            flags_enabled.append(flag)
    return flags_enabled

# Issuer address to query for no freeze status
issuer_addr = "rUpy3eEg8rqjqfUoLeBnZkscbKbFsKXC3v"

# Build account line query
print(f"Checking if the 'No Freeze' flag is enabled for the address {issuer_addr}")
acc_info = AccountInfo(account=issuer_addr, ledger_index="validated")

# Submit query
response = client.request(acc_info)

# Parse response for result
result = response.result

# Query result for no freeze status
if "account_data" in result:
    if "Flags" in result["account_data"]:
        if "lsfNoFreeze" in parse_account_root_flags(result["account_data"]["Flags"]):
            print("No Freeze is enabled")
        else:
            print("No Freeze is disabled")
